File: 2019/Alth-Py/src/Q1932.py
def solution(row, column, sum):
    max = 0
    sum += triangle[row][column]

    print(f'[{row}, {column}] checked. sum = {sum}')
    print('='*10)

    if row + 1 != triangleSize:
        for i in range(0, 2):
            result = solution(row+1, column+i, sum)
            if result > max:
                max = result
    else:
        return sum

    return max

triangle = []
triangleSize = int(input())
for i in range(triangleSize):
    line = input()
    triangle.append([int(i) for i in line.split()])

result = solution(0, 0, 0)
print(f"결과 : {result}")

File: 2019/Alth/src/Q1932.java
import java.util.LinkedList;

public class Q1932 extends Question {
    private int triangleSize, nodeSize;
    private int[][] triangle;
    private boolean[] checked;
    private int[] values;

    @Override
    public void input() {
        triangleSize = scanner.nextInt();
        scanner.nextLine();

        for (int i = 0; i < scanner.nextLine(); i++)

        for (int[] arr : triangle) {
            arr = new int[triangleSize];

            for (int i = 0; i < arr.length; i++) {
                arr[i] = 0;
            }
        }

        for (int i = 0; i < nodeSize; i++) {
            checked[i] = false;
        }

        for (int i = 0; i < triangleSize; i++) {
            String[] arr = scanner.nextLine().split(" ");

            for (String n : arr) {
                values(Integer.parseInt(n));
            }
        }

        System.out.println(triangle[3].toString());
    }

    @Override
    void init() {


        nodeSize = triangleSize * triangleSize;
        triangle = new int[triangleSize][];
        checked = new boolean[nodeSize];
        values = new int[nodeSize];
    }

    @Override
    public void solution() {

    }
}
